flexmeasures/ui/templates/crud/asset.html

```diff
@@ -369,6 +483,436 @@ <h3>All child assets for {{ asset.name }}</h3>
 <script src="https://cdn.jsdelivr.net/npm/leaflet@1.7.1/dist/leaflet-src.min.js"></script>
 <script src="{{ url_for('flexmeasures_ui.static', filename='js/map-init.js') }}"></script>
 
+<script>
+    let units = {{available_units | tojson}};
+    const selectElement = document.getElementById('unitsSelect');
+
+    units.forEach((unit, index) => {
+        const option = document.createElement('option');
+        option.value = unit;
+        option.text = unit;
+        selectElement.appendChild(option);
+    });
+</script>
+
+<script>
+    const formModal = document.getElementById('sensorsToShowForm');
+    const apiSensorsListElement = document.getElementById("apiSensorsList");
+    const spinnerElement = document.getElementById('spinnerElement');
+    let rawJSON = "{{ asset.sensors_to_show | safe}}";
+    rawJSON = rawJSON.replace(/'/g, '"');
+    let data = JSON.parse(rawJSON);
```

Comment:
This needs a more speaking name, for readability and to avoid clashes with other code!

Reply:
ill refactor that